src/atomate2/abinit/schemas/task.py

```diff
@@ -13,17 +13,97 @@
-from atomate2.abinit.utils.common import LOG_FILE_NAME, MPIABORTFILE
+from atomate2.abinit.utils.common import LOG_FILE_NAME, MPIABORTFILE, OUTPUT_FILE_NAME
 from atomate2.utils.datetime import datetime_str
-from atomate2.utils.path import get_uri, strip_hostname
+from atomate2.utils.path import get_uri
 
 logger = logging.getLogger(__name__)
 
 
+class AbinitPseudoDoc(BaseModel):
+    """Summary of the pseudopotentials used in an Abinit calculation.
+
+    Parameters
+    ----------
+    basename: str
+        Basename of the file.
+    type_psp: str
+        Type of the pseudopotential.
+    symbol: str
+        Symbol of the element.
+    Z: int
+        Atomic number of the element.
+    Z_val: int
+        Valence charge.
+    l_max: int
+        Maximum angular momentum.
+    md5: str
+        MD5 hash value.
+    filepath: str
+        Absolute path to the pseudopotential file.
+    repo_name: str
+        Name of the pseudopotentials repository if applicable.
+    table_name: str
+        Name of the accuracy table associated to the repository if applicable.
+    """
+
+    basename: str = Field(None, description="Name of the pseudopotential file.")
+    type_psp: str = Field(None, description="Type of the pseudopotential.")
+    symbol: str = Field(None, description="Symbol of the element.")
+    Z: int = Field(None, description="Atomic number of the element.")
+    Z_val: int = Field(None, description="Valence charge.")
+    l_max: int = Field(None, description="Maximum angular momentum.")
+    md5: str = Field(None, description="MD5 hash value.")
+    filepath: str = Field(
+        None, description="Absolute path of the pseudopotential file."
+    )
+    repo_name: Optional[str] = Field(
+        None, description="Name of the pseudopotentials repository."
+    )
+    table_name: Optional[str] = Field(
+        None, description="Name of the accuracy table of the repository."
+    )
+
+    @classmethod
+    def from_abinitpseudo(cls, abi_psp: AbinitPseudo) -> Self:
+        """Create a summary from an AbinitPseudo.
+
+        Parameters
+        ----------
+        abi_psp: AbinitPseudo
+            An Abinit pseudopotential.
+
+        Returns
+        -------
+        .AbinitPseudoDoc
+            The abinit pseudopotential summary.
+        """
+        dct_abi_psp = abi_psp.as_dict()
+        split_filepath = dct_abi_psp["filepath"].split("/")[::-1]
+        repo_name = None
+        for isplit in split_filepath:
+            if "ONCVPSP" in isplit or "ATOMPAW" in isplit:
+                repo_name = isplit
+
+        return cls(
+            basename=dct_abi_psp["basename"],
+            type_psp=dct_abi_psp["type"],
+            symbol=dct_abi_psp["symbol"],
+            Z=dct_abi_psp["Z"],
+            Z_val=dct_abi_psp["Z_val"],
+            l_max=dct_abi_psp["l_max"],
+            md5=dct_abi_psp["md5"],
+            filepath=dct_abi_psp["filepath"],
+            repo_name=repo_name,
```

Comment:
I think the naming and the hypens are a bit inconsistent. "basename" vs. "repo_name".